src/pages/index.js

```diff
@@ -1,81 +1,199 @@
+import "./index.css";
 import * as constants from "../utils/constants.js";
+import Api from "../components/Api.js";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
-import Section from "../components/Section.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupWithSubmit from "../components/PopupWithSubmit.js";
+import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
-import "./index.css";
 
-// Function to create card elements
+let cardSection;
+// API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7f5d30ae-91c6-4c4a-9270-9bbdbc1bcd50",
+    "Content-Type": "application/json",
+  },
+});
+// Rendering cards
 function createCard(data) {
-  const cardElement = new Card(data, constants.cardSelector, handleImageClick);
+  const cardElement = new Card(
+    data,
+    constants.cardSelector,
+    handleImageClick,
+    handleLikeClick,
+    handleDeleteClick
+  );
   return cardElement.getView();
 }
-// Section for rendering cards
-const cardSection = new Section(
-  {
-    items: constants.initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
-
-function renderCard(item) {
-  const cardElement = createCard(item);
-  cardSection.addItem(cardElement);
-}
-// UserInfo
+// UserInfo and Cards
 const userInformation = new UserInfo({
   name: ".profile__title",
   description: ".profile__description",
+  avatar: ".profile__image",
 });
 
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  // UserInfo
+  .then(([userData, cards]) => {
+    userInformation.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    });
+    userInformation.setAvatar(userData.avatar);
+    // Cards
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const cardEl = createCard(cardData);
+          cardSection.addItem(cardEl);
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+// PopupWithImage and image preview
+const popupImage = new PopupWithImage("#modal-preview");
+popupImage.setEventListeners();
+
+function handleImageClick(data) {
+  popupImage.open(data);
+}
+// Likes
+function handleLikeClick(cardId) {
+  if (cardId._isLiked) {
+    api
+      .removeLike(cardId._id)
+      .then(() => {
+        cardId._renderLikes();
```

Comment:
`_renderLikes`
`_isLiked`

Please, delete the low dash (`underscore`) from methods if you want to use them outside a class because the low dash shows a private method which is used only inside the class.